prow/bugzilla/client.go

```diff
@@ -225,11 +278,15 @@ func cloneBugStruct(bug *Bug, comments []Comment) *BugCreate {
 // clone handles the bz client calls for the bug cloning process and allows us to share the implementation
 // between the real and fake client to prevent bugs from accidental discrepencies between the two.
 func clone(c Client, bug *Bug) (int, error) {
+	subcomponents, err := c.GetSubComponentsOnBug(bug.ID)
```

Comment:
Will this error on a bugzilla instance that does not have subcomponents?

Reply:
Yes, I need to fix that. The server doesn't return an error, but it returns the standard empty bugzilla struct (which should result in `SubComponents` = `nil`, which would result in an error in the function).

Reply:
That being said, I'm not even sure if this works on non-Red Hat bugzilla at all at this point. I tried running this on other bugzillas, and those seem to return `component` as a `string` (matching what the documentation says it should be) vs. the `[]string` that the Red Hat bugzilla server returns, meaning that we can't unmarshal the json from other bugzilla servers...